MapboxCoreNavigation/RouteController.swift

```diff
@@ -40,9 +40,10 @@ public protocol RouteControllerDelegate: class {
 
      - parameter routeController: The route controller that discarded the location.
      - parameter location: The location that was discarded
+     - return: If `true`, the location is discarded and the `RouteController` will not consider it. If `false`, the location will not be thrown out.
      */
     @objc(routeController:didDiscardLocation:)
-    optional func routeController(_ routeController: RouteController, didDiscard location: CLLocation)
+    optional func routeController(_ routeController: RouteController, didDiscard location: CLLocation) -> Bool
```

Comment:
This is awkward:

* Delegator: I did discard this location!
* Delegate: No.
* Delegator: OK, on second thought, I did not discard the location!

Instead, let’s replace this method with `routeController(_:shouldMoveTo:)`:

* Delegator: Should I move to this location?
* Delegate: No.
* Delegator: OK, I won’t move to this location.